### PR TITLE
Add a test to validate hostfxr C API

### PR DESCRIPTION
**Pull request by am11**
Add a test for https://github.com/dotnet/runtime/pull/109763 and align the style as headers in `src/native/minipal/*.h`.

### CODE REVIEW

**Comment by am11** [comment]
e.g. without the struct keyword, it failed to build tests

```
  In file included from /Users/am11/projects/runtime5/src/native/corehost/test/mockhostfxr/test_c_api.c:8:
  /Users/am11/projects/runtime5/src/native/corehost/hostfxr.h:419:20: error: must use 'struct' tag to refer to type 'hostfxr_initialize_parameters'
    419 |     /*opt*/ const  hostfxr_initialize_parameters* parameters,
        |                    ^
        |                    struct 
  1 error generated.
  make[2]: *** [test/mockhostfxr/CMakeFiles/mockhostfxr_5_0.dir/test_c_api.c.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  [ 73%] Building CXX object fxr/standalone/CMakeFiles/hostfxr.dir/__/corehost_init.cpp.o
  make[1]: *** [test/mockhostfxr/CMakeFiles/mockhostfxr_5_0.dir/all] Error 2
```

File: src/native/corehost/hostfxr.h [hunk 1]
```diff
@@ -322,6 +323,10 @@ struct hostfxr_dotnet_environment_info
     const struct hostfxr_dotnet_environment_framework_info* frameworks;
 };
 
+typedef void(HOSTFXR_CALLTYPE* hostfxr_get_dotnet_environment_info_result_fn)(
+    const struct hostfxr_dotnet_environment_info* info,
+    void* result_context);
```
Comment (am11) [inline src/native/corehost/hostfxr.h:328]: Had to move it after the `struct hostfxr_dotnet_environment_framework_info` is defined, to fix:

```
  In file included from /Users/am11/projects/runtime5/src/native/corehost/test/mockhostfxr/test_c_api.c:8:
  /Users/am11/projects/runtime5/src/native/corehost/hostfxr.h:91:18: error: declaration of 'struct hostfxr_dotnet_environment_info' will not be visible outside of this function [-Werror,-Wvisibility]
     91 |     const struct hostfxr_dotnet_environment_info* info,
        |                  ^
  1 error generated.
  make[2]: *** [test/mockhostfxr/CMakeFiles/mockhostfxr_2_2.dir/test_c_api.c.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
```

**Comment by am11** [comment]
Backport together with https://github.com/dotnet/runtime/pull/110060, since it fixes another ordering issue related to C vs. C++.

**Review by elinor-fung — APPROVED** [review]
Thank you!

**Pull request merged**